test/e2e/specs/editor/plugins/format-api.spec.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Using Format API', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin( 'gutenberg-test-format-api' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin( 'gutenberg-test-format-api' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Clicking the control wraps the selected text properly with HTML code', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'First paragraph' );
+		await pageUtils.pressKeyWithModifier( 'shiftAlt', 'ArrowLeft' );
+		await editor.clickBlockToolbarButton( 'More' );
+		const button = await page.locator(
+			`//button[contains(text(), '${ 'Custom Link' }')]`
+		);
+		await button.click();
```

Comment:
Have you read the [best practice](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md#use-accessible-selectors) guide? I think we can refactor this into role selector instead:

```suggestion
		await page.click( 'role=button[name="Custom Link"i]' );
```